src/main/java/org/apache/sling/feature/analyser/task/impl/contentpackage/ArchiveValidationContextImpl.java

```diff
@@ -69,4 +70,19 @@ public Path getPackageRootPath() {
     public Collection<PackageInfo> getDependenciesPackageInfo() {
         return Collections.emptyList();
     }
+
+    @Override
+    public Object setAttribute(String name, Object value) {
```

Comment:
You can instead leverage the newly exposed class `org.apache.jackrabbit.vault.validation.context.ArchiveValidationContext` instead of implementing this interface on your own.

Reply:
The context attributes are used from https://github.com/apache/jackrabbit-filevault/blob/c475237b2f529429a47ad029791561f7d9a6413c/vault-validation/src/main/java/org/apache/jackrabbit/vault/validation/spi/impl/OverlappingFilterValidatorFactory.java#L57.

Reply:
Good pointer, although I feel that the use of ``org.apache.jackrabbit.vault.validation.context.ArchiveValidationContext`` complicates the situation, as it also requires a ``DependencyResolver``, which I couldn't find out howto create it.

Reply:
Implementing a working dependency resolver shouldn't be too hard though when just extending https://github.com/apache/jackrabbit-filevault/blob/master/vault-validation/src/main/java/org/apache/jackrabbit/vault/validation/context/AbstractDependencyResolver.java (https://issues.apache.org/jira/browse/JCRVLT-774)

Reply:
I added a ``NoopDependencyValidator`` for the moment.

Reply:
@kwin is this what you were aiming for? (Excluding resolving the proper dependencies ... that's for another ticket.)